### PR TITLE
feat: [SRM-10245]: Fixes for appD template

### PR DESCRIPTION
**Pull request by deepesh-ui**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by deepesh-ui** [comment]
retrigger jest

**Comment by deepesh-ui** [comment]
trigger cypress

**Comment by deepesh-ui** [comment]
retrigger jest

File: src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.utils.ts [hunk 1]
```diff
@@ -554,23 +558,45 @@ export const initializeNonCustomFields = (appDynamicsData: AppDynamicsData) => {
...
+  appDTier: string | SelectOption,
+  tierOptions: SelectOption[]
+) =>
   tierLoading || !appDTier
     ? { label: '', value: '' }
-    : tierOptions.find((item: SelectOption) => item.label === appDTier) || {
-        label: appDTier,
-        value: appDTier
-      }
+    : tierOptions.find((item: SelectOption) => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.utils.ts:588]: above function `setAppDynamicsApplication` doing the same operation as `setAppDynamicsTier`. Is it possible to extract the common logic to a separate function?

File: src/modules/85-cv/pages/health-source/connectors/AppDynamics/Components/AppDynamicsTier/AppDynamicsTier.tsx [hunk 2]
```diff
@@ -33,27 +33,42 @@ export default function AppDynamicsTier({
...
       className={css.tierDropdown}
       name={'appDTier'}
       label={getString('cv.healthSource.connectors.AppDynamics.trierLabel')}
       selectItems={tierOptions}
       placeholder={getPlaceholder(tierLoading, 'cv.healthSource.connectors.AppDynamics.tierPlaceholder', getString)}
       multiTypeInputProps={{
-        onChange: async item => {
+        allowableTypes: allowedTypes,
+        multitypeInputValue: multitypeInputValue,
+        onChange: async (item, _type, multiType) => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/AppDynamics/Components/AppDynamicsTier/AppDynamicsTier.tsx:61]: why `async` is used here?

**Review by praneshg239 — APPROVED** [review]

**Comment by deepesh-ui** [comment]
trigger cypress

File: src/modules/72-templates-library/components/TemplateStudio/MonitoredServiceTemplateCanvas/MonitoredServiceTemplateCanvas.tsx [hunk 3]
```diff
@@ -14,6 +14,8 @@ import type { MonitoredServiceForm } from '@cv/pages/monitored-service/component
 import type { JsonNode } from 'services/template-ng'
 import { TemplateContext } from '../TemplateContext/TemplateContext'
 import type { TemplateFormRef } from '../TemplateStudio'
+// eslint-disable-next-line
+import { MonitoredServiceProvider } from '@cv/pages/monitored-service/MonitoredServiceContext'
```
Comment (AshwinBhatkal-Harness) [inline src/modules/72-templates-library/components/TemplateStudio/MonitoredServiceTemplateCanvas/MonitoredServiceTemplateCanvas.tsx:18]: This should be moved into templates or some other higher module level. If it doesn't fit, then we may have to refactor that component accordingly. Disabling es-lint defeats the purpose of module layering.

Comment (vkbansal) [reply src/modules/72-templates-library/components/TemplateStudio/MonitoredServiceTemplateCanvas/MonitoredServiceTemplateCanvas.tsx:18]: Don't disable eslint errors. They are enforced for a reason.

**Review by vkbansal — CHANGES_REQUESTED** [review]

File: src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.tsx [hunk 4]
```diff
@@ -80,24 +82,33 @@ export default function AppDMonitoredSource({
...
-      offset: 0,
-      pageSize: 10000,
-      filter: ''
+    error: applicationError,
+    refetch: refetchAppDApplication
+  } = useGetAppDynamicsApplications({ lazy: true })
+
+  useEffect(() => {
+    if (
+      getMultiTypeFromValue(connectorIdentifier) !== MultiTypeInputType.RUNTIME &&
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.tsx:96]: please store `getMultiTypeFromValue(connectorIdentifier)` in a variable and use that for comparison.

File: src/modules/72-templates-library/components/TemplateStudio/MonitoredServiceTemplateCanvas/MonitoredServiceTemplateCanvas.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/72-templates-library/components/TemplateStudio/MonitoredServiceTemplateCanvas/MonitoredServiceTemplateCanvas.tsx:13]: resolved

File: src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.tsx (hunk 4 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/AppDynamics/AppDHealthSource.tsx]: resolved

**Review by vkbansal — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**